From 47ede33a369a3a79c84d7dcae4d3fe6dc1a934ae Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Tue, 22 Oct 2024 23:54:50 +0200
Subject: [PATCH 2/2] clang NO_CHARCONV

---
 server/TracyPrint.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/TracyPrint.hpp b/server/TracyPrint.hpp
index d38245e..dacfa70 100644
--- a/server/TracyPrint.hpp
+++ b/server/TracyPrint.hpp
@@ -16,7 +16,7 @@
 #  define NO_CHARCONV
 #endif
 
-#ifdef __GNUC__
+#if defined(__GNUC__) || defined(__clang__)
 #  define NO_CHARCONV
 #endif
 
-- 
2.47.0.windows.1

