From ea83540166375c7d025eaed7c88a06722c157391 Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Fri, 21 Mar 2025 14:11:46 +0100
Subject: [PATCH 6/9] msvc add intrin.h include

---
 source/core/slang-byte-encode-util.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/core/slang-byte-encode-util.h b/source/core/slang-byte-encode-util.h
index 81fc0ef..bf0dfbc 100644
--- a/source/core/slang-byte-encode-util.h
+++ b/source/core/slang-byte-encode-util.h
@@ -3,6 +3,10 @@
 
 #include "slang-list.h"
 
+#if SLANG_VC
+#include <intrin.h>
+#endif
+
 namespace Slang
 {
 
-- 
2.49.0.windows.1

