From f43ae71d4a4016c14be62ee77bef73ba6bb23978 Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Fri, 21 Mar 2025 16:03:48 +0100
Subject: [PATCH 7/9] change MSVC to WIN32

---
 cmake/SlangTarget.cmake | 64 ++++++++++++++++++++---------------------
 1 file changed, 31 insertions(+), 33 deletions(-)

diff --git a/cmake/SlangTarget.cmake b/cmake/SlangTarget.cmake
index eae5cf3..1b25549 100644
--- a/cmake/SlangTarget.cmake
+++ b/cmake/SlangTarget.cmake
@@ -249,45 +249,43 @@ function(slang_add_target dir type)
     endif()
 
     if(generate_split_debug_info)
-        if(MSVC)
+        if(CMAKE_SYSTEM_NAME MATCHES "Windows")
             set_target_properties(
                 ${target}
                 PROPERTIES
                     COMPILE_PDB_NAME "${target}"
                     COMPILE_PDB_OUTPUT_DIRECTORY "${output_dir}"
             )
+        elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+            # macOS - use dsymutil with --flat to create separate debug file
+            add_custom_command(
+                TARGET ${target}
+                POST_BUILD
+                COMMAND
+                    dsymutil --flat $<TARGET_FILE:${target}> -o
+                    $<TARGET_FILE:${target}>.dwarf
+                COMMAND chmod 644 $<TARGET_FILE:${target}>.dwarf
+                COMMAND ${CMAKE_STRIP} -S $<TARGET_FILE:${target}>
+                WORKING_DIRECTORY ${output_dir}
+                VERBATIM
+            )
         else()
-            if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
-                # macOS - use dsymutil with --flat to create separate debug file
-                add_custom_command(
-                    TARGET ${target}
-                    POST_BUILD
-                    COMMAND
-                        dsymutil --flat $<TARGET_FILE:${target}> -o
-                        $<TARGET_FILE:${target}>.dwarf
-                    COMMAND chmod 644 $<TARGET_FILE:${target}>.dwarf
-                    COMMAND ${CMAKE_STRIP} -S $<TARGET_FILE:${target}>
-                    WORKING_DIRECTORY ${output_dir}
-                    VERBATIM
-                )
-            else()
-                add_custom_command(
-                    TARGET ${target}
-                    POST_BUILD
-                    COMMAND
-                        ${CMAKE_OBJCOPY} --only-keep-debug
-                        $<TARGET_FILE:${target}> $<TARGET_FILE:${target}>.dwarf
-                    COMMAND chmod 644 $<TARGET_FILE:${target}>.dwarf
-                    COMMAND
-                        ${CMAKE_STRIP} --strip-debug $<TARGET_FILE:${target}>
-                    COMMAND
-                        ${CMAKE_OBJCOPY}
-                        --add-gnu-debuglink=$<TARGET_FILE:${target}>.dwarf
-                        $<TARGET_FILE:${target}>
-                    WORKING_DIRECTORY ${output_dir}
-                    VERBATIM
-                )
-            endif()
+            add_custom_command(
+                TARGET ${target}
+                POST_BUILD
+                COMMAND
+                    ${CMAKE_OBJCOPY} --only-keep-debug
+                    $<TARGET_FILE:${target}> $<TARGET_FILE:${target}>.dwarf
+                COMMAND chmod 644 $<TARGET_FILE:${target}>.dwarf
+                COMMAND
+                    ${CMAKE_STRIP} --strip-debug $<TARGET_FILE:${target}>
+                COMMAND
+                    ${CMAKE_OBJCOPY}
+                    --add-gnu-debuglink=$<TARGET_FILE:${target}>.dwarf
+                    $<TARGET_FILE:${target}>
+                WORKING_DIRECTORY ${output_dir}
+                VERBATIM
+            )
         endif()
     endif()
 
@@ -536,7 +534,7 @@ function(slang_add_target dir type)
             set(debug_dest ${library_subdir})
         endif()
 
-        if(MSVC)
+        if(WIN32)
             set(debug_file $<TARGET_PDB_FILE:${target}>)
         else()
             set(debug_file "$<TARGET_FILE:${target}>.dwarf")
-- 
2.49.0.windows.1

