From 80f59bb96f9c7209a41f257ee9c88707923d12c1 Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Fri, 21 Mar 2025 16:14:36 +0100
Subject: [PATCH 8/9] msvc include crtdgb.h

---
 source/slangc/main.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/slangc/main.cpp b/source/slangc/main.cpp
index 2090b04..e6d9830 100644
--- a/source/slangc/main.cpp
+++ b/source/slangc/main.cpp
@@ -17,6 +17,10 @@ using namespace Slang;
 #define MAIN main
 #endif
 
+#if defined(_MSC_VER)
+#include <crtdbg.h>
+#endif
+
 static void _diagnosticCallback(char const* message, void* /*userData*/)
 {
     auto stdError = StdWriters::getError();
-- 
2.49.0.windows.1

