From 774d8ee9cdc14be627c7af7b86718f3b8d1a31e0 Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Fri, 21 Mar 2025 16:29:42 +0100
Subject: [PATCH 9/9] temp lld patch

---
 cmake/CompilerFlags.cmake | 27 +++++++++++++++++++--------
 1 file changed, 19 insertions(+), 8 deletions(-)

diff --git a/cmake/CompilerFlags.cmake b/cmake/CompilerFlags.cmake
index b35f98b..aef06b5 100644
--- a/cmake/CompilerFlags.cmake
+++ b/cmake/CompilerFlags.cmake
@@ -152,14 +152,25 @@ function(set_default_compile_options target)
 
     add_supported_cxx_flags(${target} PRIVATE ${warning_flags})
 
-    add_supported_cxx_linker_flags(
-        ${target}
-        PRIVATE
-        # Don't assume that symbols will be resolved at runtime
-        "-Wl,--no-undefined"
-        # No reason not to do this? Useful when using split debug info
-        "-Wl,--build-id"
-    )
+    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+        add_supported_cxx_linker_flags(
+            ${target}
+            PRIVATE
+            # Don't assume that symbols will be resolved at runtime
+            "-Wl,--no-undefined"
+            # No reason not to do this? Useful when using split debug info
+            "-Wl,--build-id"
+        )
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        add_supported_cxx_linker_flags(
+            ${target}
+            PRIVATE
+            # Don't assume that symbols will be resolved at runtime
+            "-Wl,-undefined,error"
+            # No reason not to do this? Useful when using split debug info
+            "-Wl,-build-id"
+        )
+    endif()
 
     set_target_properties(
         ${target}
-- 
2.49.0.windows.1

