diff --git a/slang.h b/slang.h
index 54466ca..cc3a29a 100644
--- a/slang.h
+++ b/slang.h
@@ -23,7 +23,15 @@ Most applications should not need to touch this section.
 Compiler defines, see http://sourceforge.net/p/predef/wiki/Compilers/
 NOTE that SLANG_VC holds the compiler version - not just 1 or 0
 */
-#    if defined(_MSC_VER)
+#    if defined(__clang__)
+#        define SLANG_CLANG 1
+#    elif defined(__SNC__)
+#        define SLANG_SNC 1
+#    elif defined(__ghs__)
+#        define SLANG_GHS 1
+#    elif defined(__GNUC__) /* note: __clang__, __SNC__, or __ghs__ imply __GNUC__ */
+#        define SLANG_GCC 1
+#    elif defined(_MSC_VER)
 #        if _MSC_VER >= 1900
 #            define SLANG_VC 14
 #        elif _MSC_VER >= 1800
@@ -37,14 +45,6 @@ NOTE that SLANG_VC holds the compiler version - not just 1 or 0
 #        else
 #            error "unknown version of Visual C++ compiler"
 #        endif
-#    elif defined(__clang__)
-#        define SLANG_CLANG 1
-#    elif defined(__SNC__)
-#        define SLANG_SNC 1
-#    elif defined(__ghs__)
-#        define SLANG_GHS 1
-#    elif defined(__GNUC__) /* note: __clang__, __SNC__, or __ghs__ imply __GNUC__ */
-#        define SLANG_GCC 1
 #    else
 #        error "unknown compiler"
 #    endif
