diff --git a/source/slang/CMakeLists.txt b/source/slang/CMakeLists.txt
index 11a4518..768aa2a 100644
--- a/source/slang/CMakeLists.txt
+++ b/source/slang/CMakeLists.txt
@@ -151,7 +151,7 @@ target_include_directories(
 #
 
 set(SLANG_LOOKUP_GENERATOR_INPUT_JSON
-    "${VCPKG_INCLUDE_DIR}/spirv/unified1/extinst.glsl.std.450.grammar.json"
+    "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include/spirv/unified1/extinst.glsl.std.450.grammar.json"
 )
 set(SLANG_LOOKUP_GENERATOR_OUTPUT_DIR
     "${CMAKE_CURRENT_BINARY_DIR}/slang-lookup-tables/"
