diff --git a/source/slang/slang.cpp b/source/slang/slang.cpp
index e99f944..8050ef7 100644
--- a/source/slang/slang.cpp
+++ b/source/slang/slang.cpp
@@ -5706,8 +5706,7 @@ SlangResult EndToEndCompileRequest::getEntryPointCodeBlob(int entryPointIndex, i
     ComPtr<IArtifact> artifact;
     SLANG_RETURN_ON_FAIL(_getEntryPointResult(this, entryPointIndex, targetIndex, artifact));
     SLANG_RETURN_ON_FAIL(artifact->loadBlob(ArtifactKeep::Yes, outBlob));
-
-    return SLANG_E_NOT_AVAILABLE;
+    return SLANG_OK;
 }
 
 SlangResult EndToEndCompileRequest::getEntryPointHostCallable(int entryPointIndex, int targetIndex, ISlangSharedLibrary** outSharedLibrary)
@@ -5716,7 +5715,7 @@ SlangResult EndToEndCompileRequest::getEntryPointHostCallable(int entryPointInde
     ComPtr<IArtifact> artifact;
     SLANG_RETURN_ON_FAIL(_getEntryPointResult(this, entryPointIndex, targetIndex, artifact));
     SLANG_RETURN_ON_FAIL(artifact->loadSharedLibrary(ArtifactKeep::Yes, outSharedLibrary));
-    return SLANG_E_NOT_AVAILABLE;
+    return SLANG_OK;
 }
 
 SlangResult EndToEndCompileRequest::getTargetCodeBlob(int targetIndex, ISlangBlob** outBlob)
