diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6015a44..b2da0af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,6 +170,9 @@ endif()
 #
 
 find_package(Threads REQUIRED)
+find_package(glslang CONFIG REQUIRED)
+find_package(SPIRV-Tools CONFIG REQUIRED)
+find_package(SPIRV-Tools-opt CONFIG REQUIRED)
 
 if(SLANG_SLANG_LLVM_FLAVOR STREQUAL "USE_SYSTEM_LLVM")
     find_package(LLVM 13.0 REQUIRED CONFIG)
@@ -306,7 +309,7 @@ slang_add_target(
     source/slang-glslang
     MODULE
     USE_EXTRA_WARNINGS
-    LINK_WITH_PRIVATE glslang GenericCodeGen MachineIndependent SPIRV SPIRV-Tools SPIRV-Tools-opt
+    LINK_WITH_PRIVATE glslang::glslang glslang::GenericCodeGen glslang::MachineIndependent glslang::SPIRV SPIRV-Tools-static SPIRV-Tools-opt
     INSTALL
 )
 # Our only interface is through what we define in source/slang-glslang, in the
