diff --git a/clang/utils/hmaptool/CMakeLists.txt b/clang/utils/hmaptool/CMakeLists.txt
index 2891c29..5301fbb 100644
--- a/clang/utils/hmaptool/CMakeLists.txt
+++ b/clang/utils/hmaptool/CMakeLists.txt
@@ -4,11 +4,8 @@ if(CLANG_INSTALL_HMAPTOOL)
   install(PROGRAMS hmaptool DESTINATION "${CLANG_TOOLS_INSTALL_DIR}" COMPONENT hmaptool)
 endif()
 
-add_custom_target(hmaptool ALL DEPENDS "hmaptool")
-set_target_properties(hmaptool PROPERTIES FOLDER "Clang/Utils")
-
 if(CLANG_INSTALL_HMAPTOOL AND NOT LLVM_ENABLE_IDE)
   add_llvm_install_targets("install-hmaptool"
-                           DEPENDS hmaptool
+                           DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/hmaptool"
                            COMPONENT hmaptool)
 endif()
