diff --git a/libshaderc_util/src/compiler.cc b/libshaderc_util/src/compiler.cc
index 525cbc5..c59a1f4 100644
--- a/libshaderc_util/src/compiler.cc
+++ b/libshaderc_util/src/compiler.cc
@@ -706,8 +706,7 @@ std::vector<uint32_t> ConvertStringToVector(const std::string& str) {
   size_t vector_length =
       (num_bytes_str + sizeof(uint32_t) - 1) / sizeof(uint32_t);
   std::vector<uint32_t> result_vec(vector_length, 0);
-  std::strncpy(reinterpret_cast<char*>(result_vec.data()), str.c_str(),
-               str.size());
+  str.copy(reinterpret_cast<char*>(result_vec.data()), num_bytes_str);
   return result_vec;
 }
 
