From b80b3b4a7b8fa9ef10d9052d18e91619d125e5d4 Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Wed, 23 Oct 2024 14:48:10 +0200
Subject: [PATCH] remove cxx14 check

---
 src/CMakeLists.txt | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 02428d5..a618a4b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -362,12 +362,6 @@ set_target_properties(Catch2 PROPERTIES
                                 VERSION ${PROJECT_VERSION}
                               SOVERSION ${PROJECT_VERSION})
 
-# require C++14
-target_compile_features(Catch2
-  PUBLIC
-    cxx_std_14
-)
-
 configure_file(
   "${SOURCES_DIR}/catch_user_config.hpp.in"
   "${PROJECT_BINARY_DIR}/generated-includes/catch2/catch_user_config.hpp"
-- 
2.47.0.windows.1

