From 8390000a8e5aaa7fbc920ae9d5fbb374852c7ebe Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Wed, 18 Feb 2026 15:14:45 +0100
Subject: [PATCH 1/2] properly set exports on windows with clang compiler

---
 layers/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/layers/CMakeLists.txt b/layers/CMakeLists.txt
index d65de61..7cbb79a 100644
--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -475,6 +475,8 @@ elseif(MINGW)
     target_sources(vvl PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/${LAYER_NAME}.def)
     target_compile_options(vvl PRIVATE -Wa,-mbig-obj)
     set_target_properties(vvl PROPERTIES PREFIX "") # remove the prefix "lib" so the manifest json "library_path" matches
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_SYSTEM_NAME MATCHES "Windows")
+    target_link_options(vvl PRIVATE -Wl,/DEF:${CMAKE_CURRENT_SOURCE_DIR}/${LAYER_NAME}.def)
 elseif(APPLE)
 
 # IOS and APPLE can both be true (FYI)
-- 
2.52.0.windows.1

