From 4dcb6c7d0e00338c1f645b49a8637469797e04a2 Mon Sep 17 00:00:00 2001
From: Daniel Johansson <1911831+djohansson@users.noreply.github.com>
Date: Wed, 16 Oct 2024 14:06:45 +0200
Subject: [PATCH 2/2] fix detection of clang-cl

---
 tests/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index fbc4fc3..7e934db 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -67,7 +67,7 @@ if (WIN32)
     endif()
 
     # Silence errors found in clang-cl
-    if(CMAKE_C_COMPILER_ID STREQUAL "Clang" AND "${CMAKE_C_SIMULATE_ID}" MATCHES "MSVC")
+    if(CMAKE_C_COMPILER_ID STREQUAL "Clang" AND AND CMAKE_C_COMPILER_FRONTEND_VARIANT MATCHES "MSVC")
         target_compile_options(detours PRIVATE -Wno-sizeof-pointer-memaccess -Wno-microsoft-goto -Wno-microsoft-cast)
     endif()
 endif()
-- 
2.47.0.windows.1

